core/src/task.cpp

```diff
@@ -296,9 +296,17 @@ moveit_msgs::msg::MoveItErrorCodes Task::execute(const SolutionBase& s) {
 	}
 
 	auto result_future = ac->async_get_result(goal_handle);
-	if (rclcpp::spin_until_future_complete(node, result_future) != rclcpp::FutureReturnCode::SUCCESS) {
-		RCLCPP_ERROR(node->get_logger(), "Get result call failed");
-		return error_code;
+	while (result_future.wait_for(std::chrono::milliseconds(10)) != std::future_status::ready) {
+		if (pimpl()->preempt_requested_) {
+			auto cancel_future = ac->async_cancel_goal(goal_handle);
+			if (rclcpp::spin_until_future_complete(node, cancel_future) != rclcpp::FutureReturnCode::SUCCESS) {
```

Comment:
why do we need to spin here? I thought that there would be an executor spinning the node already, no?